programs/zk-token-proof-tests/tests/process_transaction.rs

```diff
@@ -1065,15 +1065,14 @@ async fn test_verify_proof_without_context<T, U>(
 
     let client = &mut context.banks_client;
     let payer = &context.payer;
-    let recent_blockhash = context.last_blockhash;
 
     // verify a valid proof (wihtout creating a context account)
     let instructions = vec![proof_instruction.encode_verify_proof(None, success_proof_data)];
     let transaction = Transaction::new_signed_with_payer(
         &instructions.with_max_compute_unit_limit(),
         Some(&payer.pubkey()),
         &[payer],
-        recent_blockhash,
+        client.get_latest_blockhash().await.unwrap(),
```

Comment:
*minor*

While this is just tests, why request the blockhash over and over again?
Considering that the current version is already caching it, it seems easier to just update the line that was requesting the blockhash in the first place.

```suggestion
    let payer = &context.payer;
    let recent_blockhash = client.get_latest_blockhash().await.unwrap();

    // verify a valid proof (wihtout creating a context account)
    let instructions = vec![proof_instruction.encode_verify_proof(None, success_proof_data)];
    let transaction = Transaction::new_signed_with_payer(
        &instructions.with_max_compute_unit_limit(),
        Some(&payer.pubkey()),
        &[payer],
        recent_blockhash,
```

Reply:
it hit https://buildkite.com/anza/agave/builds/5647#018ff290-ded1-4ce0-a248-ef109173cc6a/74-6193

<img width="811" alt="Screenshot 2024-06-13 at 11 00 23" src="https://github.com/anza-xyz/agave/assets/8209234/234e1e49-41f6-4e55-8f97-30b01019b4fb">

the line number changed but it is this func:

https://github.com/anza-xyz/agave/blob/da36ce7abebc595931295894922c47d79baae832/runtime/src/bank.rs#L3146-L3153

I found the age is increasing more than before so it failed. you should able reproduce the error by

1. clone this PR
2. revert [d368614](https://github.com/anza-xyz/agave/pull/1693/commits/d36861468a18f5dba81d5be2a362e9c7eb6f98e1)
3. `cargo test -p solana-zk-token-proof-program-tests -- --exact test_withdraw`

also I think it's harmless, and even better, to always create a tx with the latest blockhash 🤔

Reply:
Let's move it into a separate PR then.
With a proper explanation as to why this change is being made.